client/rest-high-level/build.gradle

```diff
@@ -66,6 +66,9 @@ dependencies {
   testCompile "org.elasticsearch:rest-api-spec:${version}"
 
   restSpec "org.elasticsearch:rest-api-spec:${version}"
+  // Needed for serialization tests: 
+  // (In order to serialize a server side class to a client side class or the other way around)
+  testCompile "org.elasticsearch.plugin:x-pack-core:${version}"
```

Comment:
This is a test dependency but I think it's still one that we have no precedent for. 
Maybe it would be better to have these tests somewhere in `x-pack/qa` ?

Reply:
If these tests were integration tests then I would agree. However since these tests are unit tests, I think they should stay next to the code that these tests are testing.

Reply:
I think a test dep on xpack is perfectly fine WRT unit tests here. I agree w/ @martijnvg but thank you for having the eagle eye @atorok and bringing this up, just because it is definitely a major dependency.